lightning/backend/local.go

```diff
@@ -0,0 +1,488 @@
+func NewLocalBackend(ctx context.Context, tls *common.TLS, pdAddr string, regionSplitSize int64, filePath string) (Backend, error) {
+	pdCli, err := pd.NewClient([]string{pdAddr}, tls.ToPDSecurityOption())
+	if err != nil {
+		return MakeBackend(nil), errors.Annotate(err, "construct pd client failed")
+	}
+	allStores, err := pdCli.GetAllStores(ctx, pd.WithExcludeTombstone())
+	if err != nil {
+		return MakeBackend(nil), errors.Annotate(err, "get all stores failed")
+	}
+	clients := &grpcClis{
+		clis: make(map[uint64]sst.ImportSSTClient),
+	}
+
+	tlsConf, err := tls.TransToTlsConfig()
+	if err != nil {
+		return MakeBackend(nil), err
+	}
+	splitCli := split.NewSplitClient(pdCli, tlsConf)
+
+	for _, store := range allStores {
+		// create new connection for every store
+		conn, err := grpc.DialContext(ctx, store.GetAddress(), tls.ToGRPCDialOption())
+		if err != nil {
+			return MakeBackend(nil), errors.Annotatef(err, "connect to store failed: %s", store.Address)
+		}
+		clients.clis[store.GetId()] = sst.NewImportSSTClient(conn)
+	}
+	return MakeBackend(&local{
+		grpcClis: clients,
+		engines:  make(map[uuid.UUID]localFile),
+		splitCli: splitCli,
+
+		filePath:        filePath,
+		regionSplitSize: regionSplitSize,
+
+		mutationPool: sync.Pool{New: func() interface{} { return &sst.Mutation{} }},
+	}), nil
+}
+
+// Close the importer connection.
+func (local *local) Close() {
+	local.mu.Lock()
+	defer local.mu.Unlock()
+	for _, e := range local.engines {
+		e.Close()
+	}
+}
+
+func (local *local) RetryImportDelay() time.Duration {
+	return defaultRetryBackoffTime
+}
+
+func (local *local) MaxChunkSize() int {
+	// 31 MB. hardcoded by importer, so do we
+	return 31 << 10
+}
+
+func (local *local) ShouldPostProcess() bool {
+	return true
+}
+
+func (local *local) OpenEngine(ctx context.Context, engineUUID uuid.UUID) error {
+	dbPath := path.Join(local.filePath, engineUUID.String())
+	db, err := leveldb.OpenFile(dbPath, nil)
+	if err != nil {
+		return err
+	}
+	local.mu.Lock()
+	defer local.mu.Unlock()
+	local.engines[engineUUID] = localFile{db: db, length: 0, ranges: make([]Range, 0)}
+	return nil
+}
+
+func (local *local) CloseEngine(ctx context.Context, engineUUID uuid.UUID) error {
+	// Do nothing since we will do prepare jobs in importEngine, just like tikv-importer
+	return nil
+}
+
+func (local *local) getImportClient(peer *metapb.Peer) (sst.ImportSSTClient, error) {
+	local.grpcClis.mu.Lock()
+	defer local.grpcClis.mu.Unlock()
+	cli, ok := local.grpcClis.clis[peer.GetStoreId()]
+	if !ok {
+		return nil, errors.Errorf("could not find grpc client for peer id %d", peer.GetId())
+	}
+	return cli, nil
+}
+
+func (local *local) WriteToTiKV(
+	ctx context.Context,
+	meta *sst.SSTMeta,
+	ts uint64,
+	region *split.RegionInfo,
+	mutations []*sst.Mutation) (metas []*sst.SSTMeta, err error) {
+	leader := region.Leader
+	if leader == nil {
+		leader = region.Region.GetPeers()[0]
+	}
+
+	cli, err := local.getImportClient(leader)
```

Comment:
the KV pairs should be written to every peer in the region, not just the leader.